include/base/dof_map.h

```diff
@@ -600,11 +600,7 @@ class DofMap : public ReferenceCountedObject<DofMap>,
    */
   bool has_blocked_representation() const
   {
-#ifdef LIBMESH_ENABLE_BLOCKED_STORAGE
     return ((this->n_variable_groups() == 1) && (this->n_variables() > 1));
```

Comment:
Is this correct?  If we didn't enable blocked storage then we're not using a blocked representation even if we *could*, right?

Reply:
Right. PETSc MAT takes the block size not necessary for storage, and the block size is useful for some preconditioners  such fieldsplit and HMG.

Reply:
I think I see.  So with this PR we end up calling ```MatSetBlockSize(_mat,something_bigger_than_1)```, and that's useful even if we don't set the type to ```MATBAIJ``` as well?

I guess that's ideal, but then we need to document the new behavior better.  It's probably too late to change ```has_blocked_representation()``` to ```could_have_blocked_representation_but_maybe_doesnt()``` but at least we should point out in the comment above that this function returns whether the matrix *could* be blocked, not whether it *is*, lest someone mistakenly assume it's a quick way to test whether we're using MATBAIJ or MATAIJ.

Reply:
>MatSetBlockSize(_mat,something_bigger_than_1), and that's useful even if we don't set the type to MATBAIJ as well?

Exactly right.

Reply:
MATBAIJ and MATAIJ take the block size. We should not have this restriction at the first place.

One idea is to leave these old interfaces alone.  I am going to change something in petsc_matrix.C

Like this:

```
PetscInt blocksize  = static_cast<PetscInt>(this->_dof_map-> n_variables());

ierr = MatSetBlockSize(_mat, blocksize);
```

Reply:
Oh, but that would be definitely wrong.  ```n_variables()``` is only a viable block size when every variable is of the same order and same subdomain restriction and same type.

I think the changes here are the ideal way to go, I just think we ought to change the comments to correspond to the change in the behavior.